overlord/state/state.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2022 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package state
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/juju/errors"
+)
+
+type State struct {
+	db *sql.DB
+}
+
+func NewState(db *sql.DB) *State {
+	return &State{
+		db: db,
+	}
+}
+
+// DB returns the underlying sql.DB.
+// It is expected to use the transaction of state, but sometimes a raw database
+// is required in some scenarios.
+func (s *State) DB() *sql.DB {
```

Comment:
Since Run/TxnBuilder can access the `db` field this method might be redundant.

Reply:
Maybe, I'll have a think about this.